### PR TITLE
Handled errors while fetching model info

### PR DESCRIPTION
**Pull request by vikasagrawal-ms**
Handled failures while fetching model info in download component.
Update components with latest env versions

### CODE REVIEW

**Comment by novaturient95** [comment]
LGTM. @vikasagrawal-ms, are we planning to add if-else node for if validation is successful then only go to download component and so forth?

cc: @skasturi

**Comment by vikasagrawal-ms** [comment]
> 

Plan is to make pipeline succeed if model exists in the registry. Right now we are throwing error if model fails and this is decreasing our Success Rate

**Comment by skasturi** [comment]
@vikasagrawal-ms @novaturient95 We can use conditional. Please sync with @rjaincc and let us include it as part of the pipeline component.

File: assets/training/model_management/src/run_model_download.py [hunk 1]
```diff
@@ -47,7 +48,12 @@ def validate_if_model_exists(model_id):
             model_id = re.sub(NEGATIVE_MODEL_NAME_PATTERN, "-", model_id)
             logger.info(f"Updated model_name = {model_id}")
 
-        models = ml_client_registry.models.get(name=model_id, label="latest")
+        try:
+            models = ml_client_registry.models.get(name=model_id, label="latest")
+        except Exception as e:
+            logger.warning(f"Model with name - {model_id} is not available. Error: {e}")
```
Comment (skasturi) [inline assets/training/model_management/src/run_model_download.py:54]: Could we also attempt obtaining by version & `max` to get the latest one?

File: assets/training/model_management/src/run_model_download.py [hunk 2]
```diff
@@ -47,7 +48,12 @@ def validate_if_model_exists(model_id):
             model_id = re.sub(NEGATIVE_MODEL_NAME_PATTERN, "-", model_id)
             logger.info(f"Updated model_name = {model_id}")
 
-        models = ml_client_registry.models.get(name=model_id, label="latest")
+        try:
+            models = ml_client_registry.models.get(name=model_id, label="latest")
```
Comment (novaturient95) [inline assets/training/model_management/src/run_model_download.py:52]:
```suggestion
            model = ml_client_registry.models.get(name=model_id, label="latest")
```

File: assets/training/model_management/src/run_model_download.py [hunk 3]
```diff
@@ -47,7 +48,12 @@ def validate_if_model_exists(model_id):
...
+        except Exception as e:
+            logger.warning(f"Model with name - {model_id} is not available. Error: {e}")
+            continue
+
         logger.info(f"models: {models}")
```
Comment (novaturient95) [inline assets/training/model_management/src/run_model_download.py:57]:
```suggestion
        logger.info(f"model: {model}")
```

File: assets/training/model_management/src/run_model_download.py [hunk 4]
```diff
@@ -47,7 +48,12 @@ def validate_if_model_exists(model_id):
...
         if models:
             version = models.version
```
Comment (novaturient95) [inline assets/training/model_management/src/run_model_download.py:59]:
```suggestion
            version = model.version
```

**Review by novaturient95 — APPROVED** [review]

File: assets/common/components/mlflow_model_local_validation/spec.yaml [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
+version: 0.0.4
 type: command
 
 is_deterministic: True
 
 display_name: MLFlow model local validation
 description: Validates if a MLFLow model can be loaded on a compute and is usable for inferencing.
 
-environment: azureml://registries/azureml/environments/python-sdk-v2/versions/6
+environment: azureml://registries/azureml/environments/python-sdk-v2/versions/7
```
Comment (novaturient95) [inline assets/common/components/mlflow_model_local_validation/spec.yaml:12]: Should we update all other components that has version 6 in the env?

Comment (vikasagrawal-ms) [reply assets/common/components/mlflow_model_local_validation/spec.yaml:12]: all 5 components are updated

File: assets/training/model_management/src/run_model_download.py (hunk 2 above)
Comment (vikasagrawal-ms) [reply assets/training/model_management/src/run_model_download.py:52]: ack

File: assets/training/model_management/src/run_model_download.py (hunk 3 above)
Comment (vikasagrawal-ms) [reply assets/training/model_management/src/run_model_download.py:57]: ack

File: assets/training/model_management/src/run_model_download.py (hunk 4 above)
Comment (vikasagrawal-ms) [reply assets/training/model_management/src/run_model_download.py:59]: ack

File: assets/training/model_management/src/run_model_download.py (hunk 1 above)
Comment (vikasagrawal-ms) [reply assets/training/model_management/src/run_model_download.py:54]: Ack

**Review by HrishikeshGeedMS — APPROVED** [review]

**Review by novaturient95 — APPROVED** [review]

**Pull request merged**